backend/src/main/java/ai/verta/modeldb/cron_jobs/CleanUpEntitiesCron.java

```diff
@@ -0,0 +1,121 @@
+import java.util.Calendar;
+import java.util.List;
+import java.util.TimerTask;
+import javax.persistence.OptimisticLockException;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+public class CleanUpEntitiesCron extends TimerTask {
+  private static final Logger LOGGER = LogManager.getLogger(CleanUpEntitiesCron.class);
+  private final RoleService roleService;
+  private final Integer recordUpdateLimit;
+
+  public CleanUpEntitiesCron(RoleService roleService, Integer recordUpdateLimit) {
+    this.roleService = roleService;
+    this.recordUpdateLimit = recordUpdateLimit;
+  }
+
+  /** The action to be performed by this timer task. */
+  @Override
+  public void run() {
+    LOGGER.info("CleanUpEntitiesCron wakeup");
+
+    ModelDBUtils.registeredBackgroundUtilsCount();
+    try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
+      // Update project timestamp
+      cleanProjects(session);
+    } catch (Exception ex) {
+      if (ex instanceof StatusRuntimeException) {
+        StatusRuntimeException exception = (StatusRuntimeException) ex;
+        if (exception.getStatus().getCode().value() == Code.PERMISSION_DENIED_VALUE) {
+          LOGGER.warn("CleanUpEntitiesCron Exception: {}", ex.getMessage());
+        } else {
+          LOGGER.warn("CleanUpEntitiesCron Exception: ", ex);
+        }
+      } else {
+        LOGGER.warn("CleanUpEntitiesCron Exception: ", ex);
+      }
+    } finally {
+      ModelDBUtils.unregisteredBackgroundUtilsCount();
+    }
+    LOGGER.info("CleanUpEntitiesCron finish tasks and reschedule");
+  }
+
+  private void cleanProjects(Session session) {
+    LOGGER.trace("Project cleaning");
+    String alias = "pr";
+    String deleteProjectsQueryString =
+        new StringBuilder("FROM ")
+            .append(ProjectEntity.class.getSimpleName())
+            .append(" ")
+            .append(alias)
+            .append(" WHERE ")
+            .append(alias)
+            .append(".")
+            .append(ModelDBConstants.CREATED)
+            .append(" = :created ")
+            .append(" AND ")
+            .append(alias)
+            .append(".")
+            .append(ModelDBConstants.DATE_CREATED)
+            .append(" < :created_date ")
+            .toString();
+
+    // Time less then a minute because possible to have create project request running when cron job
+    // running
+    long time = Calendar.getInstance().getTimeInMillis() - 60000; // 1 minute lesser time
+    Query projectDeleteQuery = session.createQuery(deleteProjectsQueryString);
+    projectDeleteQuery.setParameter("created", false);
+    projectDeleteQuery.setParameter("created_date", time);
+    projectDeleteQuery.setMaxResults(this.recordUpdateLimit);
+    List<ProjectEntity> projectEntities = projectDeleteQuery.list();
+
+    List<String> projectIds = new ArrayList<>();
```

Comment:
Everything below here seems to be the same as the deletion cronjob, right?

Reply:
Yup. but just added new line  `roleService.deleteProjectResources(projectIds);` new before deleting project entities from DB

Reply:
Let's avoid code duplication. Is it possible for us to re-use the code?

Reply:
No, we can not re-use it because there is some code to update the status of the sub-entities like experiments while deleting projects.

Reply:
Could you expand? The code really seems like it's mostly duplicate, so I don't understand what's the issue.

Reply:
https://github.com/VertaAI/modeldb/blob/d291a71eee766bed8381a61b28535395bb7efce7/backend/src/main/java/ai/verta/modeldb/cron_jobs/DeleteEntitiesCron.java#L136

Reply:
I still don't understand the issue, so pointing to a piece of code that I also am not familiar with doesn't really help me to understand. But sure, we can add the code duplication.